### ISSUE TITLE
Aggregate CNN variable importance scores across test set

### ISSUE THREAD

**Issue by ayushnoori** [pytorch/captum#620]
Hi there,

I have trained a CNN on an input dataset of multichannel images (where each channel represents a protein marker, for example) and am looking to compute aggregate variable importance scores (perhaps via an occlusion based method?) for each channel across my entire test set rather than on the individual image level.

Is there a supported or recommended approach in Captum for doing this? Is it as simple as averaging the integrated gradients per channel?

Thanks for your advice.

Regards,

Ayush

**Comment by bilalsal** [comment]
Hi @ayushnoori ,

If I understand correctly, you want to find out which of your k markers are more important over the entire test set?
In that case, you could use Captum's [FeatureAblation](https://captum.ai/api/feature_ablation.html) by defining k ablation masks, each of which masks one marker with a suited baseline (zeros, random values, etc).

Hope this helps

**Comment by ayushnoori** [comment]
Thanks so much @bilalsal, will look into it and report back if I have any questions.

**Comment by AlessandradellaFazia** [comment]
Hi, in the end what technique did you use?  FeatureAblation or another one?  thanks in advance
